Update coding standards to 8.2

franmomu opened the pull request:
This tries to use `doctrine/coding-standard` 8, to be able to allow php 8.

Tests should pass after merging https://github.com/doctrine/DoctrineMongoDBBundle/pull/659 that fixes some deprecations.

greg0ire committed:
Mark test not performing assertions as such

It is considered risky not to perform assertions.

greg0ire committed:
Mark test that triggers deprecation as legacy

This test loads a class that uses deprecated phpdoc syntax, but that's
intended: we want to work that the type is still guessed properly for
those too.

franmomu committed:
Update coding standards to 8.2

franmomu commented:
I cherry-picked the commits (I'll remove them if necessary), it always nice to see it green 🟢

SenseException commented:
@greg0ire The cherry-picked commits might cause problems when both branches get merged. If that's the case I suggest to remove the commits and rebase later when your changes are in the base branch.

greg0ire commented on .github/workflows/coding-standards.yml:
```diff
@@ -39,7 +39,7 @@ jobs:
 
       - name: "Install dependencies with Composer"
-        run: "composer install --no-interaction --no-progress --no-suggest"
+        run: "composer update --no-interaction --no-progress --prefer-dist --prefer-stable"
```
This file seems quite a bit out of sync with [upstream](https://github.com/doctrine/.github/blob/main/workflow-templates/coding-standards.yml), let's use the same solution instead?

franmomu replied:
Now I remembered that I tried, getting this:

https://github.com/doctrine/DoctrineMongoDBBundle/runs/1787972990?check_suite_focus=true

As the minimum stability is `dev`, it installs [`dev-master` of `slevomat/coding-standard`](https://github.com/doctrine/DoctrineMongoDBBundle/runs/1787972990?check_suite_focus=true#step:4:69) which removes [`SlevomatCodingStandard.Classes.UnusedPrivateElements`](https://github.com/slevomat/coding-standard/commit/655524a1a70b6b1cc0f75bb5c1b7c6b8286dad86).

The `dev` stability was added in https://github.com/doctrine/DoctrineMongoDBBundle/pull/602 to apparently test dev versions in travis and it was not added in #660, so maybe a solution could be to remove this `dev` stability.

greg0ire replied:
Yes, adding a step that does `composer config minimum-stability stable` could be a solution.

greg0ire commented:
> The cherry-picked commits might cause problems when both branches get merged.

What kind of problems? In my experience cherry-picking does not usually cause issues.

franmomu committed:
Ignore PHPUnit's result cache

franmomu committed:
Sync coding-standards GA job with upstream

It has been updated according to https://github.com/doctrine/.github/blob/2c60b2f943cf08df21be3dc733f1b92db8d24beb/workflow-templates/coding-standards.yml
and also the composer minimum stability has been changed for this
job in order to install only stable versions.
